sdk/client/src/client.rs

```diff
@@ -209,6 +211,36 @@ impl Client {
         .await
     }
 
+    pub async fn get_diem_id_domain_map(&self) -> Result<HashMap<String, AccountAddress>> {
+        let mut domain_map = HashMap::new();
+        let account = self.get_account(treasury_compliance_account_address()).await?.into_inner();
+        if let Some(account) = account {
+            if let AccountRoleView::TreasuryCompliance {  diem_id_domain_events_key } = account.role {
+                let mut event_index = 0;
+                let batch_size = 100;
+                if let Some(key) = diem_id_domain_events_key {
+                    loop {
+                        let domain_events = self.get_events(key, event_index, batch_size).await?.into_inner();
+                        for event in domain_events.iter() {
+                            if let EventDataView::DiemIdDomain { removed, domain, address, .. } = event.clone().data {
+                                if removed {
+                                    domain_map.remove(&domain.as_str().to_string());
+                                } else {
+                                    domain_map.insert(domain.as_str().to_string(), address);
+                                }
+                            }
+                        }
+                        if domain_events.len() < batch_size as usize {
+                            break;
+                        }
+                        event_index += batch_size;
+                    }
+                }
+            };
+        }
+        return Ok(domain_map);
+    }
```

Comment:
is there a way to create a test for this func?

Reply:
The test in main.rs for domain will test this because the tests there will call this func, for both add and remove